### DISCUSSION TITLE
아이템 41. 정의하려는 것이 타입이라면 마커 인터페이스를 사용하라

### DISCUSSION THREAD

**Discussion by corock** [Study-2-Effective-Java/Effective-Java#111] [3. 과제]
## 요약

- 마커 인터페이스와 마커 애너테이션은 각자의 용도가 존재
- 새로 추가하는 메서드 없이 단지 타입 정의가 목적이라면 마커 인터페이스를 선택
- 다음 두 가지에 해당한다면 마커 애너테이션 선택
    - 클래스나 인터페이스 외의 프로그램 요소에 마킹해야 할 때
    - 애너테이션을 적극 활용하는 프레임워크의 일부로 그 마커를 편입시키고자 할 때

## _marker interface_ > _marker annotation_

1. 마커 애너테이션과는 달리, 마커 인터페이스는 이를 구현한 클래스의 인스턴스들을 구분하는 타입으로 쓸 수 있음
2. 적용 대상을 더 정밀하게 지정할 수 있음

## _marker interface_ < _marker annotation_

거대한 애너테이션 시스템의 지원을 받고 있음

## 사용 판단 기준

> "이 마킹이 된 객체를 매개변수로 받는 메서드를 작성할 일이 있는가?"

- Maybe? 마커 인터페이스
- NEVER! 마커 애너테이션

**Comment by JoisFe** [comment]
저번에 배웠던 #42 cloneable 인터페이스도 마커 인터페이스 였죠?
